packages/location-management/src/components/LocationUnitAddEdit/Form.tsx

```diff
@@ -83,26 +83,45 @@ function removeEmptykeys(obj: any) {
   });
 }
 
+/** Function to find geolocation of a node by passing its id
+ *
+ * @param {Array<ParsedHierarchyNode>} tree Array of ParsedHierarchyNodes to find node from
+ * @param {string} id id of the node
+ * @returns {number | null} return geolocation if found else return null
+ */
+export function findParentGeoLocation(tree: ParsedHierarchyNode[], id: string): number | null {
+  const map: (number | null)[] = tree.flatMap((node) => {
+    if (node.id === id) return node.node.attributes.geographicLevel;
+    else if (node.children) return findParentGeoLocation(node.children, id);
+    else return null;
+  });
+
+  const filter = map.filter((e) => e != null);
+  return filter[0];
+}
+
```

Comment:
before : we were fetching new one from the server based on id and get the whole node as well as its children.
now : we are using this function to fetch the node from the parsed hierarchy tree and then return the geolocation if found else return null.

we don't have straight forward implementation because we have to search though out the whole tree for the node